tests/drivers/gpio/gpio_basic_api/src/test_gpio_port.c

```diff
@@ -1,21 +1,22 @@
 /*
  * Copyright (c) 2019 Nordic Semiconductor ASA
+ * Copyright 2024 NXP
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
-
 #include "test_gpio.h"
 
 #define ALL_BITS ((gpio_port_value_t)-1)
 
-static const struct device *const dev = DEVICE_DT_GET(DEV);
+static const struct device *const dev_in = DEVICE_DT_GET(DEV_IN);
+static const struct device *const dev_out = DEVICE_DT_GET(DEV_OUT);
```

Comment:
I think we should first understand why it was done this way (use `git blame` to find PR that introduced it) and if it is ok to change it.
Also, it should be done in a dedicated commit explaining all this

Reply:
it is Peter Bigot add this original with  PR #73638 with commit #ae7a59eae4b877b1990c576b307f185c3a369484
```
test/drivers/gpio: add tests for new API

Test that the new port API functions all behave as expected, including
physical vs logical level for input and output as well as masked and
set-based output operations.  Also tests the new pin API functions.

For running on real hardware this test now uses a local test-specific
devicetree binding.  For build-only tests any platform with a GPIO
alias should be tested.

The new code increases flash requirements so add a filter to exclude
platforms that won't link.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>
```

but there is not related comments

Reply:
> Also, it should be done in a dedicated commit explaining all this

updated